pkg/model/components/addonmanifests/dnscontroller/remap.go

```diff
@@ -0,0 +1,99 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package dnscontroller
+
+import (
+	"fmt"
+
+	"github.com/blang/semver/v4"
+	corev1 "k8s.io/api/core/v1"
+	addonsapi "k8s.io/kops/channels/pkg/api"
+	"k8s.io/kops/pkg/kubemanifest"
+	"k8s.io/kops/pkg/model"
+	"k8s.io/kops/pkg/model/iam"
+)
+
+// Remap remaps the dns-controller addon
+func Remap(context *model.KopsModelContext, addon *addonsapi.AddonSpec, objects []*kubemanifest.Object) error {
+	if !context.UseServiceAccountIAM() {
+		return nil
+	}
+
+	if addon.KubernetesVersion != "" {
+		versionRange, err := semver.ParseRange(addon.KubernetesVersion)
+		if err != nil {
+			return fmt.Errorf("cannot parse KubernetesVersion=%q", addon.KubernetesVersion)
+		}
+
+		if !kubernetesRangesIntersect(versionRange, semver.MustParseRange(">= 1.19.0")) {
+			// Skip; this is an older manifest
```

Comment:
This logic needs to be in sync with the iam_builder logic that decides whether to give the privileges to the service account or node

Reply:
I _think_ this is OK;  this is saying "only try to patch the latest versions of the manifests".  There is a deeper issue though, which is how to enable this on an existing cluster ... I talk about that in the top level comment.

Reply:
If the Kubernetes version is 1.11 but the feature flag is on, then iam_builder will withhold the permissions from the node role, yet this won't inject the projected volume. So dns-controller won't have the permissions.

Reply:
Right - there is "caveat emptor" with feature flags, I see you've got it with your suggested change though so I'm going to incorporate that!

Reply:
This check can probably be removed now.